CreateAR.Commons.Unity.Async/AsyncToken.cs

```diff
@@ -197,6 +199,27 @@ public IAsyncToken<TR> Map<TR>(Func<T, TR> map)
 
             return output;
         }
+        
+        /// <inheritdoc cref="IAsyncToken{T}"/>
+        public Task<T> AsTask()
+        {
+            return Task.Run(() =>
+            {
+                while (!_aborted && _resolution == null) { }
```

Comment:
This part concerns me a bit. Isn't it possible for this to lead to a deadlock? Typically, you'd see some sort of interval defined so the task explicitly releases control. Could that be passed in as a parameter with some default value?

Reply:
I don't think it could deadlock, but if the token never resolves it would leave a zombie task... Which yeah I don't think would be cleaned up by GC if the token isn't referenced anymore.

I'll add a timeout :)